### DISCUSSION TITLE
REST APIs with Flask and Python v2

### DISCUSSION THREAD

**Discussion by jslvtr** [schoolofcode-me/rest-api-sections#67] [Announcements]
It's finally here! [REST APIs with Flask and Python v2](https://go.tecla.do/rest-apis-sale) is live.

I know that many of you have been waiting a while for this since our initial announcement, #32, back in July 2021 (I still can't believe that's over a year ago).

I'm proud to say this is the best course we've ever made, and I'm really happy with the amount of content, the conciseness of the videos, the quality, and really with everything! 

The course covers:

- Flask
- Docker
- Flask-Smorest
- SQLAlchemy and Flask-SQLAlchemy (incl. many-to-many relationships)
- Flask-JWT-Extended
- Alembic and Flask-Migrate
- Git
- Deployments to Render.com

There is one more section that I want to add, but it may have to wait a bit since it's a more difficult one. That's the Celery section. I'm still not sure whether to cover Celery, rq, or something else.

Now for some deviations from the original post.

## Deployments to AWS

I really don't like Amazon, and although it is by far the most popular deployment tool out there, I decided to not cover AWS in this course. Instead, there's a section on Render.com which is a fantastic small business. Its offering is great, the pricing is good, and it's super easy to use. It's a great alternative to Heroku as well.

Other options like Google and Azure were considered, but, like AWS, were too complicated. Maybe in the future we'll add this as a separate section or a separate short series in the blog or YouTube channel.

## Flask-RESTX

After much investigation, I decided to use Flask-Smorest instead of Flask-RESTX. They're pretty similar, but Flask-Smorest uses marshmallow, which felt like a better option than Flask-RESTX's own solution.

## Docker

✅  Covered! Both for local development, with and without volumes, and for deployment.

## Flask v2

✅  Added. Although there aren't very many different things at this level, it's good to stay on top of latest versions.

## Celery

Not added yet, but it's in the works. This will be released with v2.1.

## Other libraries / content that you suggested

A few of you suggested other content to be included in the course, like websockets and OpenID. I'm still not sure about including this in the course, but now more than before, I am strongly considering it.

After re-writing, re-recording, and thoroughly editing the material, the course has ended up being substantially shorter than v1. So now there's a bit more scope for adding extra content.

## Conclusion

That's it! Thank you very much for your patience this last year, and I hope you'll enjoy the new content!

Any questions or comments, please post below. And if you find any issues / bugs, please submit a PR. Thank you!

**Comment by GeoEducator** [comment]
Awesome, how do I get started

**Reply by jslvtr** [reply]
If you already have the course on Udemy, you can just go to the course page and all the new content will be there.

If you don't have the course yet, you can buy it here: [https://go.tecla.do/rest-apis-sale](https://go.tecla.do/rest-apis-sale).

Thanks!

**Comment by anyfactor** [comment]
This is awesome. I REALLY enjoyed version 1. What is your opinion on Fast API?

I used to use Restful Flask thanks to your course, but I have recently moved to Fast API as I thought Flask is Flask, it has it's own place, the rest API spinoff versions may not be reliable enough anymore.

**Reply by jslvtr** [reply]
Hello! Thank you very much for your kind words!

FastAPI is just another library, like Flask, so just because it has "API" in the name doesn't make it better than Flask for any specific thing (I say this in response to you saying "API spinoff").

The big problem with FastAPI is that it is basically unmaintained, with 1.1k open issues and 400+ open pull requests.

In comparison, Flask is doing an absolutely fantastic job. Plus recently the owners of Flask (Pallets) added Quart into the libraries they maintain, which is an async version of Flask. I expect to see more features from that coming into Flask in the coming months.

In addition to this, the Flask ecosystem is just brilliant. Extensions aren't "spinoffs", but truly useful packages that add to Flask. There's _lots_ of them, so you can find Flask extensions to do most things you'd want to do. Flask-Smorest is an example, which actually closely resembles FastAPI in the way development works!

**Reply by anyfactor** [reply]
You are insightful as always, Jose. I will give Flask-Smorest a shot in my next project.

Thank you :)
